src/middlewares/auth.middleware.js

```diff
@@ -0,0 +1,88 @@
+import { check, validationResult } from 'express-validator';
+// import passport from 'passport';
+
+import Responses from '../utils/response.utils';
+import UserUtils from '../utils/user.utils';
+
+const loginError = { message: 'Incorrect email or password' };
+
+/**
+ * Middleware for the auth routes
+ */
+export default class AuthenticationMiddleware {
+  /**
+   * @param {ServerRequest} req
+   * @param {ServerResponse} res
+   * @param {callback} next
+   * @returns {ServerResponse} response
+   */
+  static async verifyUserToken(req, res, next) {
+    try {
+      let token;
+
+      if (req.headers.authorization) {
+        // eslint-disable-next-line prefer-destructuring
+        token = req.headers.authorization.split(' ')[1];
+      } else if (req.headers['x-access-token']) {
+        token = req.headers['x-access-token'];
+      } else if (req.headers.token) {
+        token = req.headers.token;
+      }
+
+      const decoded = UserUtils.verifyToken(token);
+
+      if (!token) {
+        return Responses.getUnauthorizedErrorResponse(res, 'No token provided.');
+      }
+
+      if (decoded.error) {
+        return Responses.getUnauthorizedErrorResponse(res, 'Failed to authenticate token.');
+      }
+      return next();
+    } catch (error) {
+      return Responses.getInternalServerErrorResponse(res, 'Internal Server Error');
```

Comment:
`Response.InternalServerError` is shorter, and in my opinion, just as good. 
Also, did you inform Oluchi that you will be making changes to her code?

Reply:
Yes, I informed her. And `Response.InternalServerError`, cool.